### PR TITLE
[6.5.x] JBPM-6653 - Use real defaults contents on new project creation

### PR DESCRIPTION
**Pull request by tsurdilo**

### CODE REVIEW

File: jbpm-designer-backend/src/main/java/org/jbpm/designer/repository/vfs/VFSRepository.java [hunk 1]
```diff
@@ -575,746 +599,144 @@ public void createGlobalDirOnNewProject( @Observes NewProjectEvent newProjectEve
...
+
+    protected static byte[] getBytesFromFile(File file) throws java.io.IOException {
+        if (file == null || file.length() > Integer.MAX_VALUE) {
+            return null;
+        }
+
+        long length = file.length();
+        byte[] bytes;
+        InputStream is = null;
+        try {
```
Comment (jhrcek) [inline jbpm-designer-backend/src/main/java/org/jbpm/designer/repository/vfs/VFSRepository.java]: You can use try-with-resources statement and avoid the finally block alltogether..
```
try (InputStream is = new FileInputStream(file)) {
 ...
}
```

File: jbpm-designer-backend/src/main/java/org/jbpm/designer/repository/vfs/VFSRepository.java [hunk 2]
```diff
@@ -575,746 +599,144 @@ public void createGlobalDirOnNewProject( @Observes NewProjectEvent newProjectEve
...
+                    logger.error("Unable to create default color themes.");
+                }
+
+                // default icons
+                try {
+                    String defaultEmailIcon = servletContext.getRealPath(designerContext + "/defaults/defaultemailicon.gif");
+                    String defaultLogIcon = servletContext.getRealPath(designerContext + "/defaults/defaultlogicon.gif");
+                    String defaultServiceIcon = servletContext.getRealPath(designerContext + "/defaults/defaultservicenodeicon.png");
+
+                    ioService.write(ioService.get(URI.create(globalDirPath + separator + "defaultemailicon.gif")),
```
Comment (jhrcek) [inline jbpm-designer-backend/src/main/java/org/jbpm/designer/repository/vfs/VFSRepository.java]: Could you please extract this "copy resource to ioservice" functionality to separate function?

You could make this much more readable by then just calling
```
List<String> iconFileNames = Arrays.asList("defaultemailincon.gif", "defautlogicon.gif", ...)
iconFileNames.forEach(this::copyToGlobalsDir);
```

File: jbpm-designer-backend/src/main/java/org/jbpm/designer/repository/vfs/VFSRepository.java (hunk 1 above)
Comment (jhrcek) [reply jbpm-designer-backend/src/main/java/org/jbpm/designer/repository/vfs/VFSRepository.java]: Second thought: you can replace the entire method by `IOUtils.toByteArray(new FileInputStream(file));`

File: jbpm-designer-backend/src/main/java/org/jbpm/designer/repository/vfs/VFSRepository.java [hunk 3]
```diff
@@ -575,746 +599,144 @@ public void createGlobalDirOnNewProject( @Observes NewProjectEvent newProjectEve
...
-                    "                \"namespace\" : \"http://b3mn.org/stencilset/bpmn2.0#\",\n" +
-                    "                \"connectingType\" : \"http://b3mn.org/stencilset/bpmn2.0#SequenceFlow\",\n" +
-                    "                \"parent\" : [\"4\"],\n" +
-                    "                \"children\" : [],\n" +
-                    "                \"xyOffset\" : [120,0]\n" +
-                    "            }\n" +
-                    "        ]\n" +
-                    "    }\n" +
-                    "]");
+            ioService.write(ioService.get(URI.create(globalDirPath + separator + "backboneformsinclude.fw")),
```
Comment (jhrcek) [inline jbpm-designer-backend/src/main/java/org/jbpm/designer/repository/vfs/VFSRepository.java]: I'd prefer to see this extracted in a method (like createEmtyFile) and then call it like
```
Arrays.asList("backbonejsinclude.fw", "cancelbutton.fw", ...)
   .forEach(this::createEmptyFile)
```

**Comment by tsurdilo** [comment]
@jhrcek yours and @jomarko suggestions incorporated here - thanks guys again!

**Pull request merged**